lib/core/utils/guild_utils.dart

```diff
@@ -0,0 +1,57 @@
+import 'dart:convert';
+
+import 'package:app/core/domain/chat/entities/guild.dart';
+import 'package:http/http.dart' as http;
+
+class GuildUtils {
+  static String guildApiEndpoint = 'https://api.guild.xyz';
+
+  static Future<Guild> getGuildDetail(num guildId) async {
```

Comment:
these logic should move to ChatRepositoryImpl , as this should belong to data layer.

Reply:
understood, but the utils should only contain helper fucntion, for those getting raw data, should let data layer cover it